types/react-codemirror/index.d.ts

```diff
@@ -1,22 +1,41 @@
-// Type definitions for React Codemirror v0.2.6
+// Type definitions for React Codemirror v1.0.0
 // Project: https://github.com/JedWatson/react-codemirror
 // Definitions by: Vicky Lai <https://github.com/velveret>
+//                 Rudi Chen <https://github.com/rudi-c>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.4
 
 /// <reference types="react"/>
 /// <reference types="codemirror"/>
 
 declare namespace ReactCodeMirror {
     interface ReactCodeMirrorProps extends React.Props<ReactCodeMirror> {
-        onChange?: (newValue: string) => any; // called when a change is made
-        onFocusChange?: (focused: boolean) => any; // called when the editor is focused or loses focus
-        onScroll?: (scrollInfo: CodeMirror.ScrollInfo) => any; // called when the editor is scrolled
-        options?: CodeMirror.EditorConfiguration; // options passed to the CodeMirror instance
-        path?: string; // the identifying name for the textarea
-        value?: string; // the editor value
-        className?: string; // CSS className for the outer element
-        codeMirrorInstance?: CodeMirror.Editor; // the CodeMirror instance
+        /** Automatically focuses the editor when it is mounted (default false) */
+        autoFocus?: boolean;
+        /** Automatically persist changes to underlying textarea (default false) */
+        autoSave?: boolean;
+        /** Adds a custom CSS class to the editor */
+        className?: string;
+        /** Provides a specific CodeMirror instance (defaults to `require('codemirror')`) */
+        codeMirrorInstance?: CodeMirror.Editor;
+        /** Provides the default (not changed tracked) value to the editor */
+        defaultValue?: string;
+        /** Set the name of the editor input field */
+        name?: string;
+        /** Called when a change is made */
+        onChange?: (newValue: string, change: CodeMirror.EditorChange) => any;
+        /** Called when the cursor is moved */
+        onCursorActivity?: (cm: CodeMirror.Editor) => any;
```

Comment:
prefer `codemirror` over `cm` to match react-codemirror docs and be more descriptive